From 67bdb372f04738d1e7f6fb323eb9d3143b4dd6f2 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Mon, 8 Jun 2020 07:17:25 +0530
Subject: [PATCH 14/25] gio/ginetaddress: Fix build on WinRT

On WinRT in6addr_any and in6addr_loopback are not available at link
time, so we initialize them with the _INIT functions in ws2ipdef.h

ginetaddress.c.obj : error LNK2001: unresolved external symbol in6addr_any
ginetaddress.c.obj : error LNK2001: unresolved external symbol in6addr_loopback
---
 gio/ginetaddress.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gio/ginetaddress.c b/gio/ginetaddress.c
index 66d8897..685a44c 100644
--- a/gio/ginetaddress.c
+++ b/gio/ginetaddress.c
@@ -33,6 +33,15 @@
 #include "glibintl.h"
 #include "gnetworkingprivate.h"
 
+#ifdef G_WINAPI_ONLY_APP
+/* On WinRT in6addr_any and in6addr_loopback are not available at link time, so
+ * we initialize them with _INIT functions in ws2ipdef.h */
+static CONST IN6_ADDR g_in6addr_any = IN6ADDR_ANY_INIT;
+static CONST IN6_ADDR g_in6addr_loopback = IN6ADDR_LOOPBACK_INIT;
+#define in6addr_any g_in6addr_any
+#define in6addr_loopback g_in6addr_loopback
+#endif
+
 struct _GInetAddressPrivate
 {
   GSocketFamily family;
-- 
2.38.1

